packages/backend/src/services/GitIntegrationService/GitIntegrationService.ts

```diff
@@ -117,36 +115,53 @@ export class GitIntegrationService extends BaseService {
         };
     }
 
-    static async createBranch({
-        owner,
-        repo,
-        mainBranch,
-        token,
-        branch,
-    }: {
-        branch: string;
-        owner: string;
-        repo: string;
-        mainBranch: string;
-        token: string;
-    }) {
-        const { sha: commitSha } = await getLastCommit({
-            owner,
-            repo,
-            branch: mainBranch,
-            token,
-        });
+    static async createBranch(gitProps: GitProps) {
+        const { owner, repo, mainBranch, token, branch, type, hostDomain } =
+            gitProps;
+
+        let commitSha: string;
+        if (type === DbtProjectType.GITHUB) {
+            const { sha } = await GithubClient.getLastCommit({
+                owner,
+                repo,
+                branch: mainBranch,
+                token,
+            });
+            commitSha = sha;
+        } else {
+            const { sha } = await GitlabClient.getLastCommit({
+                owner,
+                repo,
+                branch: mainBranch,
+                token,
+                hostDomain,
+            });
+            commitSha = sha;
+        }
+
         Logger.debug(
             `Creating branch ${branch} from ${mainBranch} (commit: ${commitSha}) in ${owner}/${repo}`,
         );
-        // create branch in git
-        const newBranch = await createBranch({
-            branch,
-            owner,
-            repo,
-            sha: commitSha,
-            token,
-        });
+
+        if (type === DbtProjectType.GITHUB) {
+            await GithubClient.createBranch({
+                branch,
+                owner,
+                repo,
+                sha: commitSha,
+                token,
+            });
+        } else {
+            await GitlabClient.createBranch({
+                branch,
+                owner,
+                repo,
+                sha: commitSha,
+                token,
+                hostDomain,
+            });
+        }
```

Comment:
[suggestion] add a `const client = this.getGitClient(type)` method to get rid of the if/else branches